### PR TITLE
Fix: Don't drop disabled cells

### LINKED ISSUE #2676: Disabled cell content sometimes disappears from file
The content of disabled cells sometimes disappear from the file
This is an issue, as it can get lost when the notebook is closed.

This has been happening "randomly" for months, so it is not related to the latest release.
But I finally found a mwe, tested with `Pluto v0.19.30` and `julia 1.9.2`.

Note: in the following video, every time the `git gui` window appears, it shows the previous state,
until I hit F5 to get the current state (still have to figure out how to show the keystrokes...)
The last cell is in a "disabled" state.
The sequance shown in this video is
1. Click on the  `using BenchmarkTools` cell, which modifies the file.
2. The modification is not visible in the `git gui` that gets focus (previous state, no diff),
    then I press F5 in the `git gui` window and the view is updated, showing the actual diff.
3. Then I click on the "save code without running" on the disabled cell,
and do the same actions in the `git gui` window to show that the file gets fixed (no diff).
4. Execute the `using BenchmarkTools` cell, which modifies the file,
and do the same actions in the `git gui` window to show that the cell content got lost gain.

https://github.com/fonsp/Pluto.jl/assets/10962809/a74bd6a5-77ea-491b-bf26-ad220c651350

Here is the notebook:
https://gist.github.com/ederag/c6628a895b10b973d7fdc184d8236232#file-losing_disabled-jl

Note that `BenchmarkTools` is _not_ the culprit. The issue is the same with ` TestEnv `
or any other package present in the global environment.

To reproduce, just run the notebook from a fresh julia session.

### PR DESCRIPTION
**Pull request by Pangoraw**
Fixes #2676.

### CODE REVIEW

File: src/notebook/saving and loading.jl [hunk 1]
```diff
@@ -56,18 +56,24 @@ function save_notebook(io::IO, notebook::Notebook)
     println(io)
 
     cells_ordered = collect(topological_order(notebook))
-    
+    # TODO: add test for this case
```
Comment (fonsp) [inline src/notebook/saving and loading.jl:59]: Can you add a comment here to explain when this can happen?

**Comment by ederag** [comment]
While in "safe preview" the file is modified (that's #2674),
but after "Run notebook code", the file is back to normal (only the Pluto version changed from 0.19.29).
So this PR does fix #2676 indeed, even for larger, more involved files.
Thanks !

**Comment by Pangoraw** [comment]
Thanks a lot for testing @ederag and for the detailed report!

**Comment by fonsp** [comment]
@Pangoraw merging this because I am doing a release, but can you still add that comment? (But don't push to `main` until the release is tagged 🙈)

**Pull request merged**